kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { useContext, useRef, useState } from 'react';
+
+import { getLocale, gettext, Interpolated } from '../l10n.js';
+import UserProvider from '../user-provider.jsx';
+
+export default function SubscriptionForm() {
+    const locale = getLocale();
+    const subscriptionFormRef = useRef(null);
+    const userData = useContext(UserProvider.context);
+    const [subscribeFormDisabled, setSubscribeFormDisabled] = useState(false);
+    const [enableSubscribeButton, setEnableSubscribeButton] = useState(false);
+
+    /**
+     * Toggles the enabled state of the subscription button
+     */
+    const enableButton = () => {
+        if (!enableSubscribeButton) {
+            setEnableSubscribeButton(true);
+        } else {
+            setEnableSubscribeButton(false);
+        }
+    };
+
+    /**
+     * Opens Stripe modal allowing a user to complete their subscription.
+     * @param {Object} event - The form submit event
+     */
+    const submit = event => {
+        event.preventDefault();
+
+        const subscriptionForm = subscriptionFormRef.current;
+        const formData = new FormData(subscriptionForm);
+
+        setSubscribeFormDisabled(true);
+
+        const stripeHandler = window.StripeCheckout.configure({
+            key: window.mdn.stripePublicKey,
+            locale: 'auto',
+            name: 'MDN Web Docs',
+            zipCode: true,
+            currency: 'usd',
+            amount: 500,
+            email: userData ? userData.email : '', // TODO: email is not currently exposed
```

Comment:
I did not include the `label` attribute here as I get the following warning in the console:

```
StripeCheckout.configure: Unrecognized option 'label'. Did you mean 'name'? You can learn about the available configuration options in the Checkout docs: https://stripe.com/docs/checkout
```